src/Container/LCParticleContainer.cpp

```diff
@@ -240,144 +308,150 @@ void LCParticleContainer::handleBoundaryAction() {
                 std::array<double, 3> v_arg = {p->getV().at(0), p->getV().at(1), -1*(p->getV().at(2))};
                 calcWithHalo(p, x_arg, v_arg);
               } else { // i == 5      //Boundary to other XY Plane
                 std::array<double, 3> x_arg = {p->getX().at(0),p->getX().at(1), cell_size.at(2)*cell_count.at(2) + bounds.at(i)};
                 std::array<double, 3> v_arg = {p->getV().at(0), -p->getV().at(1), -1*(p->getV().at(2))};
                 calcWithHalo(p, x_arg, v_arg);
               }
-              //Particle has same absolute velocity but
-              //eg. boundary is XY, then velocity in x and y are same but in z is -z;
-              //Particle has same x and y position but z is same distance from boundary but reversed
-              //Particle has no force
-
             } // if(bounds.at(i) != -1.0 && bounds.at(i) <= (pow(2, 1/6)/2)){
-
           } //reflectiv
           else if(boundary_types[i] == 3) {
-
-            //creates new Particle in OponentCell with same Atributes exept for X_Arg
-            std::array<double,3> x = findOponentXYZ(p->getX());
-            Particle oponent(x, p->getV(), p->getM(), p->getType());
-            particles.push_back(oponent);
+            if(bounds.at(i+6) >= 0.0) {
+              std::array<double,3> x = findOponentXYZ(p);
+              p->setX(x);
+            }
           } //periodic
+        }// if(boundary_types[i] == 2){
+    }// for(auto p: boundaryparticles) {
+}
 
-          }// if(boundary_types[i] == 2){
-
-      }// for(auto p: boundaryparticles) {
-
-  }
+std::array<double, 12> LCParticleContainer::getInfluencingBoundarysWithDistance(Particle * p) {
+  //returns distance to influencing Boundarys or -1 in case this Boundary isnt influencing. next 6 digits show, if Particle is ascending the Border(1) or descending (-1)
+  std::array<double, 12> result = {-1.0, -1.0, -1.0, -1.0, -1.0, -1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
+  //get position and decide which boundary is affecting the Particle in the future
+  const double x = p->getX().at(0);
+  const double y = p->getX().at(1);
+  const double z = p->getX().at(2);
 
+  const int x_id = int(floor(x / cell_size.at(0)));
+  const int y_id = int(floor(y / cell_size.at(1)));
+  const int z_id = int(floor(z / cell_size.at(2)));
 
-std::array<double, 6> LCParticleContainer::getInfluencingBoundarysWithDistance(Particle * p) {
-  std::array<double, 6> result = {-1.0, -1.0, -1.0, -1.0, -1.0, -1.0};
-  //get position and decide which boundary is affecting the Particle in the future
-  double x = p->getX().at(0);
-  double y = p->getX().at(1);
-  double z = p->getX().at(2);
-  if(x <= cell_size.at(0)){ //Particle close to boundary 1
+  if(x_id == 0){ //Particle close to boundary 1
     double distance1 = x;
     result.at(0) = distance1;
+    if(p->getV().at(0) < 0.0) {
+      result.at(6) = 1.0;
+    } else if(p->getV().at(0) > 0.0) {
+      result.at(6) = -1.0;
+    }
```

Comment:
some optimizations possible in this function. you can update result only if the condition satisfies, and leave it to default otherwise. this way you will avoid all `else ` parts